### ISSUE TITLE
Unlit : Animations get discarded since a socket is not found KHR_animation_pointer

### ISSUE THREAD

**Issue by The0Dev** [KhronosGroup/glTF-Blender-IO#2208]
**Describe the bug**
Hi. I'm sorry if it's too early to report this, but my only intention is to help. I've been testing the KHR_animation_pointer branch and noticed that, in some cases, texture offset animations get discarded because the function get_socket returns None.
` socket = get_socket(asset['blender_nodetree'], True, "Base Color")` 

More specifically, everything seems to work with regular PBR materials but it does not when the KHR_materials_unlit extension is used.

**To Reproduce**
1. Import the attached .glb files.
2. See the results.

**Expected behavior**
Both assets should be animated.

**Screenshots**
![khr_anim_pointer_blender](https://github.com/KhronosGroup/glTF-Blender-IO/assets/98645239/19615607-c065-4178-a302-44c637b88638)
![khr_anim_pointer_babylon](https://github.com/KhronosGroup/glTF-Blender-IO/assets/98645239/bb916d4d-5ea0-4a7e-b048-98b457a06155)

**.blend file/ .gltf (mandatory)**
[khr_animation_pointer_bug.zip](https://github.com/KhronosGroup/glTF-Blender-IO/files/15049508/khr_animation_pointer_bug.zip)

**Version**
 - OS: Linux
 - Blender Version: 4.2

**Additional context**
Both assets are properly displayed here:
[sandbox.babylonjs.com](https://sandbox.babylonjs.com/)

**Comment by julienduroure** [comment]
Thanks for the report, I will have a look soon

**Comment by The0Dev** [comment]
Alright! By the way, is the exporter able to fully utilize the extension? I tried checking various boxes, but I could only convert regular TRS animation to KHR_animation_pointer.

**Comment by julienduroure** [comment]
@The0Dev 

For unlit => Seems unlit is not part of extension that can have some animation, if I understand the extension specification correctly.
See https://github.com/KhronosGroup/glTF/blob/main/specification/2.0/ObjectModel.adoc#extension-pointers
More investigation in progress

KHR_animation_pointer is only available with animation "Scene" mode (documentation still need to be written) for all pointers, except for conversion of TRS, that is also available for others.

(Note that TRS conversion to pointer is broken for now on "Scene" mode

**Comment by The0Dev** [comment]
> KHR_animation_pointer is only available with animation "Scene" mode

Thank you for clarifying that. I hope the extension will become available in other modes in the future. (For instance, imagine a case where one wants a character with multiple animations to have different animated material properties in those animations)

As for unlit materials, an unlit material simply uses available values defined by its "pbr" parent material that are not related to lighting. 

`/materials/{}/pbrMetallicRoughness/baseColorTexture/extensions/KHR_texture_transform/offset` 
does not seem to be excluded from that.

>The common Unlit material is defined by adding the KHR_materials_unlit extension to any glTF material. When present, the extension indicates that a material should be unlit and use available baseColor values, alpha values, and vertex colors while ignoring all properties of the default PBR model related to lighting or color.

```json
{
    "materials": [
        {
            "name": "MyUnlitMaterial",
            "pbrMetallicRoughness": {
                "baseColorFactor": [ 0.5, 0.8, 0.0, 1.0 ]
            },
            "extensions": {
                "KHR_materials_unlit": {}
            }
        }
    ]
}
```

**Comment by julienduroure** [comment]
> (Note that TRS conversion to pointer is broken for now on "Scene" mode)

Fixed

> Thank you for clarifying that. I hope the extension will become available in other modes in the future. (For instance, imagine a case where one wants a character with multiple animations to have different animated material properties in those animations

I don't plan to work on it until the new data model for Animation in Blender is available (See Blender Baklava project)

> Unlit

Ah, yes. I probably missed this case in my implementation. Added in my TODO list

**Comment by julienduroure** [comment]
This should be solved now.
Fixed by a22d39876dfc702492595c147b447e5670196785
